client/interfaces.go

```diff
@@ -56,6 +56,7 @@ type ContainerServer interface {
 	GetContainers() (containers []api.Container, err error)
 	GetContainer(name string) (container *api.Container, ETag string, err error)
 	CreateContainer(container api.ContainersPost) (op *Operation, err error)
+	CreateContainerFromImage(source ImageServer, image api.Image, imgcontainer api.ContainersPost) (op *RemoteOperation, err error)
```

Comment:
Well, my first pass was exactly that, keep everything low level and do everything in the downstream clients, but I came to the conclusion that we realistically need at least 3 convenience functions:
 - CopyImage (already there)
 - CopyContainer (for migration)
 - CreateContainerFromImage (because it's the most common thing and quite tricky)

They're effectively the 3 code paths where you have to interact with more than one LXD, so those will be the 3 functions that give you a RemoteOperation back rather than an Operation.

The difference between RemoteOperation and Operation is that RemoteOperation can track operations on two servers and will automatically cycle between all addresses of the source server.

I also expect those 3 functions to be where we'll eventually add the "push" bits needed for push mode and for relay mode. Though right now that's only supported by containers and so should only affect CopyContainer initially.